rng/unix/getentropy.ml

```
@@ -0,0 +1,20 @@
+
+external getrandom_buf : bytes -> int -> int -> unit = "mc_getrandom" [@@noalloc]
```

I found this library: https://randombytes.cr.yp.to/security.html, which implements this ordered choice:

> librandombytes-kernel uses getrandom if it finds it, otherwise getentropy if it finds it, otherwise /dev/urandom. In the case of /dev/urandom, librandombytes-kernel waits at program startup for /dev/random to become readable; however, it skips this if the file /var/run/urandom-ready exists (or if /dev/random does not exist).

The wait on startup for /dev/random is probably not needed on modern Linuxes that wait for the RNG to be seeded (although I think very old kernels had that problem).

---

https://randombytes.cr.yp.to/librandombytes-20240318/kernel/getentropy_wrapper.c.html this has various workarounds for detecting the availability of getentropy by setting up a signal handler for SIGSYS (I assume some OS would raise a SIGSYS instead of returning ENOSYS).